addons/account_payment/controllers/portal.py

```diff
@@ -8,42 +11,123 @@
 
 class PortalAccount(portal.PortalAccount, PaymentPortal):
 
+    @http.route(['/my/invoices/overdue'], type='http', auth="public", methods=['GET'], website=True)
+    def portal_my_overdue_invoices(self, access_token=None, **kw):
+        overdue_invoices = None
+        if access_token:
+            partner_id = request.env['account.move'].sudo().search([('access_token', '=', access_token)]).partner_id.id
+            overdue_invoices = request.env['account.move'].sudo().search(self._get_overdue_invoices_domain(partner_id))
+        else:
+            try:
+                request.env['account.move'].check_access_rights('read')
+                overdue_invoices = request.env['account.move'].search(self._get_overdue_invoices_domain())
+            except (AccessError, MissingError):
+                return request.redirect('/my')
+
+        values = self._overdue_invoices_get_page_view_values(overdue_invoices, access_token, **kw)
+        if access_token and 'post_params' in values:
+            # sudo_payment signals that invoices tokens needn't be validated
+            values['post_params']['sudo_payment'] = True
```

Comment:
This triggers so many red flags :see_no_evil: 

In general, if you end up with a "should I bypass security" parameter that you carry around, it means that something is wrong in the design of either the code or the model.

For example, what if a malicious user creates a request with `sudo_payment=True`?